### ISSUE TITLE
Add abroad mode

### ISSUE THREAD

**Issue by Snake87r** [AdguardTeam/AdguardForAndroid#108]
Add a function called Abroad mode
allow only certain apps and limited traffic daily or weekly (or personalized period of time) for when you go abroad..
And add the possibility to block image, video, swf file etc..

Made the function available only for premium..

**Comment by TPS** [comment]
We in the States would call this "Roaming" mode. :+1:

**Comment by ameshkov** [comment]
> And add the possibility to block image, video, swf file etc..

It is possible already with your user filter.
You can add rules like:
```
*.jpg
*.swf
```

**Comment by ameshkov** [comment]
> Add a function called Abroad mode
> allow only certain apps and limited traffic daily or weekly (or personalized period of time) for when you go abroad..

Looks too complex to me, it will be really hard to explain how to use it.

Maybe something simpler? Like if device goes abroad/roaming Adguard blocks all background apps net access?

**Comment by TPS** [comment]
Yes!  But I understand him to also be asking for a way to mark exceptions to that — I.e., I won't want any apps but, e.g., my MMS app, & that perhaps only from 9a to 5p weekdays.

The way I understand this, it could be an evolving feature, with updates going further down the :rabbit: hole. :wink:

**Comment by ameshkov** [comment]
> But I understand him to also be asking for a way to mark exceptions to that — i.e., I won't want any apps but, e.g., my MMS app, & that perhaps only from 9a to 5p weekdays.

Don't you think that's an overkill?

I suggest something like blocking background apps and when app comes to foreground allow user to unblock it (like how it's done now when the app is blocked by the firewall).

**Comment by TPS** [comment]
Yep, that works, if you'll also add an entirely optional "Remember this" checkbox (default _off_).

I figure the point of a request like this, when users ask for the moon, start with a pebble & build from there — but don't forget to build! Moon-like timescale's perhaps ok. :wink:

**Comment by ameshkov** [comment]
@TPS 
> I figure the point of a request like this, when users ask for the moon, start with a pebble & build from there — but don't forget to build! Moon-like timescale's perhaps ok. :wink:

True:)

@Snake87r what do you think?

**Comment by TPS** [comment]
@ameshkov It just occurred to me what @Snake87r means by
> And add the possibility to block image, video, swf file etc..

He, of course, means to have such a filter turned on _only_ during roaming, when these are heavy data-use MIME-types.  I guess that's a filtering enhancement request?  Also on moon-timescale? :wink:

**Comment by ameshkov** [comment]
I just thought that it could be implemented easily IF we had Adguard "profiles". You could simply switch to another profile with necessary filters enabled.

**Comment by TPS** [comment]
Wow! That's more than I was thinking of.…  Though I do think a full profile system may not be used much, other than perhaps just a roaming/abroad vs home, or day vs night, i.e., most folks will have max 2 profiles, probably mostly only 1.

**Comment by Snake87r** [comment]
Hello,
I don't think the foreground pop-up solution is good for this use, but the profile solution is very good, so with this the user can create a roaming profile or any other profile he want.
If it's possibile the suggestion is:

In the firewall screen in the menu add this:

Change current profile
Create new profile
Edit Profile

A user create a profile and the app save (only the enabled option like enable filter ads, allow Wifi, allow mobile, with this if the user install other app are disabled by default)

Default profile (all enabled)
Save only the disabled option like (not enabled filter ads), Wifi not permitted, mobile not permitted, with this if he user install other app are enabled by default

Or a best solution, in the profile creation add an option to choose between enable new apps by default or disable new apps by default..

The swf jpg and other file filter can be added in future if it's too long to do..

PS: Excuse me for my english, I hope I've explained correctly my solution..

**Comment by ameshkov** [comment]
> The swf jpg and other file filter can be added in future if it's too long to do..

"Profile" should include all AG settings including user filter, so blocking svg/jpg and such won't be a problem.

**Comment by ameshkov** [comment]
I'll add this issue to 4.0 Milestone (when we plan to introduce user profiles and cloud sync)

**Comment by ameshkov** [comment]
merged into https://github.com/AdguardTeam/AdguardForAndroid/issues/3379

**Comment by TPS** [comment]
> I just thought that it could be implemented easily IF we had Adguard "profiles". You could simply switch to another profile with necessary filters enabled.

@ameshkov I'm very, _very_ sorry to necropost 🙇🏾‍♂️, but I must ask:  how is this implemented in, e.g., AGa v4.4n9?

The last info I see is that it's merged into 
https://github.com/AdguardTeam/AdguardForAndroid/issues/3379, where I asked @ https://github.com/AdguardTeam/AdguardForAndroid/issues/3379#issuecomment-1325973485
> Are you planning to reopen the linked issues that are _not_ implemented?

& received answer
> We'll audit the tasks on v4.0 closer to the release date and see which ones should be shifted to a later date.

Are we there yet? 😉  The reason I bring is up _now_ is that AGVPNa has the parallel request https://github.com/AdguardTeam/AdGuardVPNForAndroid/issues/350, which seems to make more sense in Integrated mode if this is implemented here, as well.

**Comment by ameshkov** [comment]
Heh, well, we had to limit ourselves a bit. Profile is a complicated concept that would just take too much of our time.

**Comment by TPS** [comment]
> Profile is a complicated concept that would just take too much of our time.

Does that mean https://github.com/AdguardTeam/AdGuardVPNForAndroid/issues/350 is also unlikely?

**Comment by ameshkov** [comment]
Yeah, probably

**Comment by TPS** [comment]
What _is_ this currently actually duplicate _to_?
